release/v1alpha1/types.go

```diff
@@ -0,0 +1,88 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PromotionTestResults aggregates the outputs of the promotion tests
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type PromotionTestResults struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec the inputs used to create the promotion test
+	Spec PromotionTestSpec `json:"spec,omitempty"`
+
+	// Status is the current status of the promotion test
+	Status PromotionTestResultsStatus `json:"status,omitempty"`
+}
+
+// PromotionTestSpec has the information to represent a PromotionTest
+type PromotionTestSpec struct {
+	//TODO: What are the required pieces for this...
+	Name string `json:"name,omitempty"`
+}
+
+// PromotionTestResultsStatus the status of all the promotion test jobs
+type PromotionTestResultsStatus struct {
+	// BlockingJobResults stores the results of all blocking jobs
+	BlockingJobResults []JobResults `json:"blockingJobResults,omitempty"`
+	// InformingJobResults stores the results of all informing jobs
+	InformingJobResults []JobResults `json:"informingJobResults,omitempty"`
+	// AnalysisJobResults stores the results of all analysis jobs
+	AnalysisJobResults []JobResults `json:"analysisJobResults,omitempty"`
+}
+
+// JobAggregateResultState the final state of the job
+type JobAggregateResultState string
+
+const (
+	// PendingJobAggregateResultState job currently running
+	PendingJobAggregateResultState JobAggregateResultState = "Pending"
+	// FailedJobAggregateResultState failed job aggregation
+	FailedJobAggregateResultState JobAggregateResultState = "Failed"
+	// SuccessJobAggregateResultState successful job aggregation
+	SuccessJobAggregateResultState JobAggregateResultState = "Success"
+)
+
+// JobResults encapsulates the name of the job, all the results of the jobs, and an aggregated
+// result of all the jobs
+type JobResults struct {
```

Comment:
ReleasePayloadStatus

Reply:
Add `Conditions []metav1.Conditions`.  I think you have overall conditions of 

1. PayloadPromotionJobsPassed
2. PayloadPromotionJobsFailed

Reply:
some jobs allow retries in case of failures.  Is the number of retries known when the JobResult is created?